app/src/main/java/com/cornellappdev/android/eatery/ui/components/general/MenuItems.kt

```diff
@@ -0,0 +1,96 @@
+package com.cornellappdev.android.eatery.ui.components.general
+
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.lazy.LazyListScope
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material.Divider
+import androidx.compose.material.Text
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.unit.dp
+import com.cornellappdev.android.eatery.data.models.MenuCategory
+import com.cornellappdev.android.eatery.data.models.MenuItem
+import com.cornellappdev.android.eatery.ui.theme.EateryBlueTypography
+import com.cornellappdev.android.eatery.ui.theme.GrayZero
+
+data class MenuCategoryViewState(
+    val category: String,
+    val items: List<MenuItemViewState>
+)
+
+fun MenuCategoryViewState.toMenuCategory() =
+    MenuCategory(category = category, items = items.map { it.toMenuItem() })
+
+data class MenuItemViewState(
+    val isFavorite: Boolean,
+    val item: MenuItem,
+)
+
+fun MenuItemViewState.toMenuItem() = item
+
+typealias ItemName = String
```

Comment:
Lol, any reason for this? I didn't even know you could do this in kotlin.

Reply:
I think it makes the parameters for `onFavoriteClick` a bit more readable, it specifies it should be an item name and not an ID or something. But it's arguable whether this actually helps, since the implementor might be confused as to what `ItemName` type actually is. Lmk if you want me to remove it.